.github/workflows/release.yml

```diff
@@ -0,0 +1,218 @@
+# Copyright 2023 DSR Corporation, Denver, Colorado.
+# https://www.dsr-corporation.com
+# SPDX-License-Identifier: Apache-2.0
+
+name: Release
+
+on:
+  push:
+    branches: [main]
+
+permissions:
+  contents: write
+
+
+jobs:
+
+  checks:
+    name: Check version & existing release
+    runs-on: ubuntu-latest
+    outputs:
+      current_version: ${{ steps.current_version.outputs.current_version }}
+      existing_release_info: ${{ steps.check_existing_release.outputs.release_info }}
+      existing_assets: ${{ steps.check_existing_release.outputs.release_assets }}
+      already_in_crates_io: ${{ steps.check_in_crates_io.outputs.already_in_crates_io != '' }}
+    defaults:
+      run:
+        working-directory: cli
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Get current version
+        id: current_version
+        run: |
+          version="$(cargo -q metadata --no-deps \
+            | jq -r '.packages[] | select(.name == "indy-cli-rs") | .version')"
+          echo current_version=$version >> $GITHUB_OUTPUT
+          echo "$version"
+        shell: bash
+
+      - name: Check existing release
+        id: check_existing_release
+        run: |
+          release_info="$(curl -s https://api.github.com/repos/${{ github.repository }}/releases \
+              | jq '.[] | select(.name == "v${{ steps.current_version.outputs.current_version }}")')"
+          
+          # Generate random delimiter as recommended in docs
+          # See https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#multiline-strings
+          delimiter="$(openssl rand -hex 8)"
+          echo "release_info<<${delimiter}" >> $GITHUB_OUTPUT
+          echo "$release_info" >> $GITHUB_OUTPUT
+          echo "${delimiter}" >> $GITHUB_OUTPUT
+          echo "existing release info: $release_info"
+
+          release_assets="$(echo "$release_info" | jq -c '[.assets[].name]')"
+          echo "release_assets=${release_assets}" >> $GITHUB_OUTPUT
+          echo "existing release assets: $release_assets"
+        shell: bash
+
+      - name: Check if already deployed to crates.io
+        id: check_in_crates_io
+        run: |
+          out="$(curl -s https://crates.io/api/v1/crates/indy-cli-rs | jq -r '.versions[] | .num' \
+            | grep '^${{ steps.current_version.outputs.current_version }}$')"
+          echo already_in_crates_io=$out >> $GITHUB_OUTPUT
+          echo "in crates.io check: $out"
+        shell: bash {0}  # to opt-out of default fail-fast behavior
+
+  create_release:
+    name: Create release if needed
+    runs-on: ubuntu-latest
+    needs: checks
+
+    steps:
+      - name: Create GitHub Release
+        if: ${{ !needs.checks.outputs.existing_release_info }}
+        uses: actions/create-release@v1
```

Comment:
`actions/create-release` is no longer maintained.  Please switch to a different action.  Indy-node and indy-plenum projects use `softprops/action-gh-release@v1`.